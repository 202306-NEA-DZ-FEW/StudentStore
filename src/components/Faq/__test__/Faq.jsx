import { useRouter } from "next/router";

export default function Faq({ t }) {
    const route = useRouter();
    return (
        <div
            className='max-w-screen-xl mx-auto px-5 mb-20 text-[#32314D]'
            dir={route.locale === "ar" ? "rtl" : "ltr"}
        >
            <div className='flex flex-col items-center'>
                <h2 className='font-bold text-4xl mt-5 text-center tracking-tight'>
                    {t("Frequently asked questions")}
                </h2>
            </div>
            <div className='grid divide-y divide-neutral-200 max-w-xl mx-auto mt-8'>
                <div className='py-5'>
                    <details className='group'>
                        <summary className='flex justify-between items-center font-medium cursor-pointer list-none'>
                            <span>{t("question1")}</span>
                            <span className='transition group-open:rotate-180'>
                                <svg
                                    fill='none'
                                    height='24'
                                    shape-rendering='geometricPrecision'
                                    stroke='currentColor'
                                    stroke-linecap='round'
                                    stroke-linejoin='round'
                                    stroke-width='1.5'
                                    viewBox='0 0 24 24'
                                    width='24'
                                >
                                    <path d='M6 9l6 6 6-6'></path>
                                </svg>
                            </span>
                        </summary>
                        <p className='text-neutral-700 mt-3 group-open:animate-fadeIn'>
                            {t("answer1")}
                        </p>
                    </details>
                </div>
                <div className='py-5'>
                    <details className='group'>
                        <summary className='flex justify-between items-center font-medium cursor-pointer list-none'>
                            <span>{t("question2")}</span>
                            <span className='transition group-open:rotate-180'>
                                <svg
                                    fill='none'
                                    height='24'
                                    shape-rendering='geometricPrecision'
                                    stroke='currentColor'
                                    stroke-linecap='round'
                                    stroke-linejoin='round'
                                    stroke-width='1.5'
                                    viewBox='0 0 24 24'
                                    width='24'
                                >
                                    <path d='M6 9l6 6 6-6'></path>
                                </svg>
                            </span>
                        </summary>
                        <p className='text-neutral-700 mt-3 group-open:animate-fadeIn'>
                            {t("answer2")}
                        </p>
                    </details>
                </div>
                <div className='py-5'>
                    <details className='group'>
                        <summary className='flex justify-between items-center font-medium cursor-pointer list-none'>
                            <span>{t("question3")}</span>
                            <span className='transition group-open:rotate-180'>
                                <svg
                                    fill='none'
                                    height='24'
                                    shape-rendering='geometricPrecision'
                                    stroke='currentColor'
                                    stroke-linecap='round'
                                    stroke-linejoin='round'
                                    stroke-width='1.5'
                                    viewBox='0 0 24 24'
                                    width='24'
                                >
                                    <path d='M6 9l6 6 6-6'></path>
                                </svg>
                            </span>
                        </summary>
                        <p className='text-neutral-700 mt-3 group-open:animate-fadeIn'>
                            {t("answer3")}
                        </p>
                    </details>
                </div>
                <div className='py-5'>
                    <details className='group'>
                        <summary className='flex justify-between items-center font-medium cursor-pointer list-none'>
                            <span>{t("question4")}</span>
                            <span className='transition group-open:rotate-180'>
                                <svg
                                    fill='none'
                                    height='24'
                                    shape-rendering='geometricPrecision'
                                    stroke='currentColor'
                                    stroke-linecap='round'
                                    stroke-linejoin='round'
                                    stroke-width='1.5'
                                    viewBox='0 0 24 24'
                                    width='24'
                                >
                                    <path d='M6 9l6 6 6-6'></path>
                                </svg>
                            </span>
                        </summary>
                        <p className='text-neutral-700 mt-3 group-open:animate-fadeIn'>
                            {t("answer4")}
                        </p>
                    </details>
                </div>
                <div className='py-5'>
                    <details className='group'>
                        <summary className='flex justify-between items-center font-medium cursor-pointer list-none'>
                            <span>{t("question5")}</span>
                            <span className='transition group-open:rotate-180'>
                                <svg
                                    fill='none'
                                    height='24'
                                    shape-rendering='geometricPrecision'
                                    stroke='currentColor'
                                    stroke-linecap='round'
                                    stroke-linejoin='round'
                                    stroke-width='1.5'
                                    viewBox='0 0 24 24'
                                    width='24'
                                >
                                    <path d='M6 9l6 6 6-6'></path>
                                </svg>
                            </span>
                        </summary>
                        <p className='text-neutral-700 mt-3 group-open:animate-fadeIn'>
                            {t("answer5")}
                        </p>
                    </details>
                </div>
            </div>
        </div>
    );
}
