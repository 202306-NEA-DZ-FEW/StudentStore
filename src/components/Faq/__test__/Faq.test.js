import renderer from "react-test-renderer";
import Faq from "./Faq";

jest.mock("next/router", () => ({
    useRouter: () => ({
        pathname: "/",
    }),
}));

const t = (key) => key;
it("renders correctly", () => {
    const tree = renderer.create(<Faq t={t} />).toJSON();
    expect(tree).toMatchSnapshot();
});
