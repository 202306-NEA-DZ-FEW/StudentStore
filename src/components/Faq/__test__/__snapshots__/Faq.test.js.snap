// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-w-screen-xl mx-auto px-5 mb-20"
  dir="ltr"
>
  <div
    className="flex flex-col items-center"
  >
    <h2
      className="font-bold text-5xl mt-5 text-center tracking-tight"
    >
      Frequently asked questions
    </h2>
  </div>
  <div
    className="grid divide-y divide-neutral-200 max-w-xl mx-auto mt-8"
  >
    <div
      className="py-5"
    >
      <details
        className="group"
      >
        <summary
          className="flex justify-between items-center font-medium cursor-pointer list-none"
        >
          <span>
            question1
          </span>
          <span
            className="transition group-open:rotate-180"
          >
            <svg
              fill="none"
              height="24"
              shape-rendering="geometricPrecision"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M6 9l6 6 6-6"
              />
            </svg>
          </span>
        </summary>
        <p
          className="text-neutral-600 mt-3 group-open:animate-fadeIn"
        >
          answer1
        </p>
      </details>
    </div>
    <div
      className="py-5"
    >
      <details
        className="group"
      >
        <summary
          className="flex justify-between items-center font-medium cursor-pointer list-none"
        >
          <span>
            question2
          </span>
          <span
            className="transition group-open:rotate-180"
          >
            <svg
              fill="none"
              height="24"
              shape-rendering="geometricPrecision"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M6 9l6 6 6-6"
              />
            </svg>
          </span>
        </summary>
        <p
          className="text-neutral-600 mt-3 group-open:animate-fadeIn"
        >
          answer2
        </p>
      </details>
    </div>
    <div
      className="py-5"
    >
      <details
        className="group"
      >
        <summary
          className="flex justify-between items-center font-medium cursor-pointer list-none"
        >
          <span>
            question3
          </span>
          <span
            className="transition group-open:rotate-180"
          >
            <svg
              fill="none"
              height="24"
              shape-rendering="geometricPrecision"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M6 9l6 6 6-6"
              />
            </svg>
          </span>
        </summary>
        <p
          className="text-neutral-600 mt-3 group-open:animate-fadeIn"
        >
          answer3
        </p>
      </details>
    </div>
    <div
      className="py-5"
    >
      <details
        className="group"
      >
        <summary
          className="flex justify-between items-center font-medium cursor-pointer list-none"
        >
          <span>
            question4
          </span>
          <span
            className="transition group-open:rotate-180"
          >
            <svg
              fill="none"
              height="24"
              shape-rendering="geometricPrecision"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M6 9l6 6 6-6"
              />
            </svg>
          </span>
        </summary>
        <p
          className="text-neutral-600 mt-3 group-open:animate-fadeIn"
        >
          answer4
        </p>
      </details>
    </div>
    <div
      className="py-5"
    >
      <details
        className="group"
      >
        <summary
          className="flex justify-between items-center font-medium cursor-pointer list-none"
        >
          <span>
            question5
          </span>
          <span
            className="transition group-open:rotate-180"
          >
            <svg
              fill="none"
              height="24"
              shape-rendering="geometricPrecision"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M6 9l6 6 6-6"
              />
            </svg>
          </span>
        </summary>
        <p
          className="text-neutral-600 mt-3 group-open:animate-fadeIn"
        >
          answer5
        </p>
      </details>
    </div>
  </div>
</div>
`;
